pylint/checkers/refactoring.py

```diff
@@ -311,6 +317,38 @@ def leave_functiondef(self, _):
         # new scope = reinitialize the stack of nested blocks
         self._nested_blocks = []
 
+    def visit_raise(self, node):
+        self._check_stop_iteration_inside_generator(node)
+
+    def _check_stop_iteration_inside_generator(self, node):
+        """
+        Check if an exception of type StopIteration is raised inside a generator
+        """
+        scope = node.scope()
+        if isinstance(scope, astroid.FunctionDef) and scope.is_generator():
+            exception_name = getattr(node.exc, 'name', '')
+            # Get all hand made exceptions
+            non_std_exc = [klass for klass in
```

Comment:
Maybe it can be reduced to something like below:
```
if any(_class is StopIteration for _class in exc.mro()):
    self.add_message()
```
It's somewhat a pseudocode, but it may be trivial to make an actual code from it.